datasource/netcdf.py

```diff
@@ -551,142 +549,126 @@ def obj_to_netcdf_id(self, object_id, normalized_object_type):
-        for v in variables:
-            # Select the source netcdf:
-            if v in self.available_subgrid_map_vars:
-                ds = self.ds
-            elif v in self.available_aggregation_vars:
-                ds = self.ds_aggregation
-            else:
-                continue
-
-            # Keep the netCDF array in memory for performance
-            if caching:
-                try:
-                    variable = self.cache[v]
-                except KeyError:
-                    try:
-                        variable = ds.variables[v][:]  # make copy
-                        self.cache[v] = variable
-                    except KeyError:
-                        log("Variable not in netCDF: %s, skipping..." % v)
-                        continue
-            else:
-                variable = ds.variables[v][:]
+        # Select the source netcdf:
+        if variable in self.available_subgrid_map_vars:
+            ds = self.ds
+        elif variable in self.available_aggregation_vars:
+            ds = self.ds_aggregation
+        else:
+            raise ValueError("Invalid variable: %s" % variable)
 
+        # Keep the netCDF array in memory for performance
+        if caching:
             try:
-                # shape ds.variables['q'] array = (t, number of ids)
-                vals = variable[:, netcdf_id]
+                data = self.cache[variable]
             except KeyError:
-                log("Variable not in netCDF: %s, skipping..." % v)
-                continue
-            except IndexError:
-                log("Id %s not found for %s" % (netcdf_id, v))
-                continue
-            timeseries_vals[v] = vals
-        return timeseries_vals
-
-    def get_values_timestamp(self, parameter, timestamp,
-                              source='default'):
-
-        v = parameter
-        if v in self.available_subgrid_map_vars:
+                data = ds.variables[variable][:]  # make copy
+                self.cache[variable] = data
+        else:
+            data = ds.variables[variable][:]
+
+        try:
+            # shape ds.variables['q'] array = (t, number of ids)
+            vals = data[:, netcdf_ids]
+        except KeyError:
+            log("Variable not in netCDF: %s" % variable)
+            raise
+        except IndexError:
+            log("Id %s not found for %s" % (netcdf_ids, variable))
+            raise
+        return vals
+
+    def get_values_by_id(self, variable, object_type, object_id, caching=True):
+        """Convenience method to get a regular array if only one ID is needed.
+        """
+        vals = self.get_values_by_ids(variable=variable,
+                                      object_type=object_type,
+                                      object_ids=[object_id],
+                                      caching=caching)
+        # Convert row array to regular array.
+        return vals[:, 0]
+
+    def get_values_by_timestamp(self, variable, timestamp_idx):

```

Comment:
Better make naming around timestep en timestamp also consistent. Suggestion:
timestep_nr = nr of output timestep starting at 0 (is equal to index in netCDF)
timestep = time in seconds between the results
timestamp = time in seconds from start of the simulation, so timestep_nr * timestep
datetime = seconds from 1-1-1970, is start_time + timestamp